src/mol-canvas3d/passes/hi-z.ts

```diff
@@ -0,0 +1,559 @@
+
+        for (const td of this.levelData) {
+            td.framebuffer.destroy();
+            td.texture.destroy();
+        }
+
+        this.levelData.length = 0;
+        for (let i = 0; i < levels; ++i) {
+            const framebuffer = this.webgl.resources.framebuffer();
+            const levelSize = Math.pow(2, levels - i - 1);
+            const size = Vec2.create(levelSize, levelSize);
+            const invSize = Vec2.create(1 / levelSize, 1 / levelSize);
+            const texture = this.webgl.resources.texture('image-float32', 'alpha', 'float', 'nearest');
+            texture.define(levelSize, levelSize);
+            texture.attachFramebuffer(framebuffer, 0);
+            this.levelData.push({ texture, framebuffer, size, invSize, offset: 0 });
+        }
+
+        let offset = 0;
+        for (let i = 0, il = levels; i < il; ++i) {
+            const td = this.levelData[i];
+            if (i >= minLevel) {
+                this.levelData[i].offset = offset;
+                offset += td.size[0];
+            }
+        }
+
+        this.clear();
+    }
+
+    setProps(props: Partial<HiZProps>) {
+        if (!this.supported) return;
+
+        if (this.props.minLevel !== props.minLevel) {
+            Object.assign(this.props, props);
+            const { x, y, width, height } = this.viewport;
+            this.setViewport(x, y, width, height);
+        } else {
+            Object.assign(this.props, props);
+            if (!this.props.enabled) this.clear();
+        }
+    }
+
+    //
+
+    private debug?: {
+        container: HTMLDivElement
+        canvas: HTMLCanvasElement
+        ctx: CanvasRenderingContext2D
+        rect: HTMLDivElement
+    };
+
+    private initDebug(element: HTMLElement) {
+        if (!element.parentElement) return;
+
+        const container = document.createElement('div');
+        Object.assign(container.style, {
+            display: 'block',
+            position: 'absolute',
+            pointerEvents: 'none',
+        });
+        element.parentElement.appendChild(container);
+
+        const canvas = document.createElement('canvas');
+        const ctx = canvas.getContext('2d');
+        if (!ctx) throw new Error('Could not create canvas 2d context');
+        Object.assign(canvas.style, {
+            width: '100%',
+            height: '100%',
+            imageRendering: 'pixelated',
+            position: 'relative',
+            pointerEvents: 'none',
+        });
+        container.appendChild(canvas);
+
+        const rect = document.createElement('div');
+        Object.assign(rect.style, {
+            display: 'none',
+            position: 'absolute',
+            pointerEvents: 'none',
+        });
+        element.parentElement.appendChild(rect);
+
+        this.debug = { container, canvas, ctx, rect };
+    }
+
+    private showRect(p: Vec4, occluded: boolean) {
+        if (!this.supported || !this.props.enabled || !this.ready || !this.debug) return;
```

Comment:
make this a single property since it's repeated quite often, maybe `get canDebug()` ?

Reply:
why not :)